### DISCUSSION TITLE
[RAGFlow] Use OSS componets instead of source avaliable counter parts

### DISCUSSION THREAD

**Discussion by http403** [infiniflow/ragflow#1258] [Ideas]
Hi community.

Since I saw RagFlow wanna to [flourishes via open-source collaboration](https://github.com/infiniflow/ragflow/blob/v0.6.0/README.md#-contributing), I wonder will the community willing to swap some of the components with source available license to FOSS alternatives to fully embrace the philosophy and spirit of open source. 

Currently I spot two: ElasticSearch to OpenSearch, and Redis to Valkey. I believe two are drop in replacements (except ElasticSearch as the SDK refuse to work with OpenSearch but I believe the code change will be minimal). 

I think I can take on the migration task but may need heavy code review.

Thanks for the consideration.

**Comment by yingfeng** [comment]
Thank you so much for the heads up and advice!

I'd like to ask a question about the Redis license, it does seem to have a lot of limitations that don't fit together in this software release. Outside of Valkey, I've also found [Dragonfly](https://github.com/dragonflydb/dragonfly) to be a good alternative, and I was wondering which one you favor?

One question about Elasticsearch is that its license prohibits providing the software to third parties as a hosted or managed Service, so it seems like using Elasticsearch in RAGflow does not have problems?

In addition, using Elasticsearch for RAGflow is only a temporary solution for now, after our own database infinity is mature and stable(during this summer), it will be the main choice for RAGflow, because it provides much more powerful features than Elasticsearch with multiple recall, much better performance, and lower resource consumption.

**Reply by http403** [reply]
Valkey. Dragonfly is licensed under BSL which is a source available license. While BSL does have change date and change license, it isn't the point. The goal is to fully embrace the philosophy and spirit of open source, which doesn't discriminate against fields of endeavor. Meaning said license shouldn't limit how the end user use/deploy/offer the software, the exact reason why I suggest to swap out Elasticsearch. However, if Elasticsearch is just a temporary solution and intent to use infinity later, then I don't see much benefit to swap. Still, if Elasticsearch will be offered as an alternative in the future, I will still suggest replacing it with OpenSearch instead.

**Reply by yingfeng** [reply]
Got it, thanks for your suggestion. Valkey is a good candidate to replace Redis.
Regarding to Elasticsearch, it will be removed in future, as long as Infinity is totally stable. However, if the community users insist on keeping Elasticsearch as a choice, it would be replaced with OpenSearch at that time.
